libres/lib/res_util/block_fs.cpp

```diff
@@ -271,92 +213,32 @@ static void file_node_init_fwrite(const file_node_type *file_node,
 */
 
 static int file_node_header_size(const char *filename) {
-    file_node_type *file_node;
-    return sizeof(file_node->status) + sizeof(file_node->node_size) +
-           sizeof(file_node->data_size) + sizeof(NODE_END_TAG) +
+    return sizeof(BlockMeta::status) + sizeof(BlockMeta::node_size) +
+           sizeof(BlockMeta::data_size) + sizeof(NODE_END_TAG) +
            sizeof(int) /* embedded by the util_fwrite_string routine */ +
            strlen(filename) + 1 /* \0 */;
 }
 
-static void file_node_set_data_offset(file_node_type *file_node,
-                                      const char *filename) {
-    file_node->data_offset =
-        file_node_header_size(filename) - sizeof(NODE_END_TAG);
-}
-
-static void file_node_dump_index(const file_node_type *file_node,
-                                 FILE *index_stream) {
-    util_fwrite_int(file_node->status, index_stream);
-    util_fwrite_long(file_node->node_offset, index_stream);
-    util_fwrite_int(file_node->node_size, index_stream);
-    util_fwrite_int(file_node->data_offset, index_stream);
-    util_fwrite_int(file_node->data_size, index_stream);
+static void file_node_set_data_offset(BlockMeta &block, const char *filename) {
+    block.data_offset = file_node_header_size(filename) - sizeof(NODE_END_TAG);
 }
 
-/*
-static file_node_type * file_node_index_fread_alloc( FILE * stream ) {
-  node_status_type status = util_fread_int( stream );
-  long int node_offset    = util_fread_long( stream );
-  int node_size           = util_fread_int( stream );
-  {
-    file_node_type * file_node = file_node_alloc( status , node_offset , node_size );
-    file_node->data_offset = util_fread_int( stream );
-    file_node->data_size   = util_fread_int( stream );
-
-    return file_node;
-  }
-}
-*/
-
-static file_node_type *file_node_index_buffer_fread_alloc(buffer_type *buffer) {
+static BlockMeta file_node_index_buffer_fread_alloc(buffer_type *buffer) {
```

Comment:
Should probably avoid `node_index` as we have the related concept `BlockMeta` I say rename to `read_block`.